### PR TITLE
3682 mailersend components

### PR DESCRIPTION
**Pull request by alysonturing**
### <samp>🤖 Generated by Copilot at fe789c4</samp>

This pull request adds a new app integration for MailerSend, a service that allows users to send and track emails. It introduces a `mailersend.app.mjs` file that defines the app properties and methods, a `common` folder that contains the base and webhook objects for the MailerSend sources, and several source files that emit events based on different email activity events. It also updates the `package.json` and `pnpm-lock.yaml` files to include the `mailersend` dependency.

### <samp>🤖 Generated by Copilot at fe789c4</samp>

> _To send and track emails with ease_
> _We added some MailerSend sources_
> _They use webhooks and `domainId`_
> _To emit events of different kinds_
> _And `mailersend.app` for resources_

### <samp>🤖 Generated by Copilot at fe789c4</samp>

*  Add `mailersend` app integration and sources for email activity events ([link](https://github.com/PipedreamHQ/pipedream/pull/6086/files?diff=unified&w=0#diff-917f95bd10b3ffaa6e1d3f2045104bd24f6ad2a465b291b63e6d7bc22ceafca6L1-R63), [link](https://github.com/PipedreamHQ/pipedream/pull/6086/files?diff=unified&w=0#diff-cbed1d38295f6b740086049a25e9ddcfb384949c844104bd77bece5a1a5ecc59R1-R28), [link](https://github.com/PipedreamHQ/pipedream/pull/6086/files?diff=unified&w=0#diff-236f564dcc8ac6c4deea44f37411817f93373d015f3fe1d820670aeb8b0caa80R1-R62), [link](https://github.com/PipedreamHQ/pipedream/pull/6086/files?diff=unified&w=0#diff-11790568a7481142be96b31886933411a6d300b421d4362f70be666ef2c6ab89R1-R20), [link](https://github.com/PipedreamHQ/pipedream/pull/6086/files?diff=unified&w=0#diff-80a6ddca9a2cf9346dc62f5fe26686442f90b70a474d703ba1039e28d8e09acbR1-R20), [link](https://github.com/PipedreamHQ/pipedream/pull/6086/files?diff=unified&w=0#diff-0e8316f69ef1662dea89ba95b5311d58710be639f752f3212d32289bdc097ae4R1-R20), [link](https://github.com/PipedreamHQ/pipedream/pull/6086/files?diff=unified&w=0#diff-0dd48af8978e611f4edb23544f3c4ca9faeede20983e6341cc87d660e888429eR1-R20), [link](https://github.com/PipedreamHQ/pipedream/pull/6086/files?diff=unified&w=0#diff-cc0172de8049b7e8b151980f383b2305b9986f04a7bb9288e93c7d0f034dcdd1R1-R20), [link](https://github.com/PipedreamHQ/pipedream/pull/6086/files?diff=unified&w=0#diff-667c7f802dd41c7174dea8840326316be3667c4bc9fd610bdcecd9e3770f2dc7R1-R20), [link](https://github.com/PipedreamHQ/pipedream/pull/6086/files?diff=unified&w=0#diff-e23cf0c540ac87e2e778de796435da82ecf3fd2ad3b8ab30dd644c057007b974R1-R20), [link](https://github.com/PipedreamHQ/pipedream/pull/6086/files?diff=unified&w=0#diff-f5f6248003441a5fedaae639e43bca687b3c052d31fd3e7d9a42e2fd8b693344R1-R20))
  * Define `domainId` property and API methods in `mailersend.app.mjs` ([link](https://github.com/PipedreamHQ/pipedream/pull/6086/files?diff=unified&w=0#diff-917f95bd10b3ffaa6e1d3f2045104bd24f6ad2a465b291b63e6d7bc22ceafca6L1-R63))
  * Create `base.mjs` and `webhook.mjs` files in `common` folder to share common logic for MailerSend sources ([link](https://github.com/PipedreamHQ/pipedream/pull/6086/files?diff=unified&w=0#diff-cbed1d38295f6b740086049a25e9ddcfb384949c844104bd77bece5a1a5ecc59R1-R28), [link](https://github.com/PipedreamHQ/pipedream/pull/6086/files?diff=unified&w=0#diff-236f564dcc8ac6c4deea44f37411817f93373d015f3fe1d820670aeb8b0caa80R1-R62))
  * Create source files for each email activity event in `mailersend` folder, inheriting from `webhook.mjs` ([link](https://github.com/PipedreamHQ/pipedream/pull/6086/files?diff=unified&w=0#diff-11790568a7481142be96b31886933411a6d300b421d4362f70be666ef2c6ab89R1-R20), [link](https://github.com/PipedreamHQ/pipedream/pull/6086/files?diff=unified&w=0#diff-80a6ddca9a2cf9346dc62f5fe26686442f90b70a474d703ba1039e28d8e09acbR1-R20), [link](https://github.com/PipedreamHQ/pipedream/pull/6086/files?diff=unified&w=0#diff-0e8316f69ef1662dea89ba95b5311d58710be639f752f3212d32289bdc097ae4R1-R20), [link](https://github.com/PipedreamHQ/pipedream/pull/6086/files?diff=unified&w=0#diff-0dd48af8978e611f4edb23544f3c4ca9faeede20983e6341cc87d660e888429eR1-R20), [link](https://github.com/PipedreamHQ/pipedream/pull/6086/files?diff=unified&w=0#diff-cc0172de8049b7e8b151980f383b2305b9986f04a7bb9288e93c7d0f034dcdd1R1-R20), [link](https://github.com/PipedreamHQ/pipedream/pull/6086/files?diff=unified&w=0#diff-667c7f802dd41c7174dea8840326316be3667c4bc9fd610bdcecd9e3770f2dc7R1-R20), [link](https://github.com/PipedreamHQ/pipedream/pull/6086/files?diff=unified&w=0#diff-e23cf0c540ac87e2e778de796435da82ecf3fd2ad3b8ab30dd644c057007b974R1-R20), [link](https://github.com/PipedreamHQ/pipedream/pull/6086/files?diff=unified&w=0#diff-f5f6248003441a5fedaae639e43bca687b3c052d31fd3e7d9a42e2fd8b693344R1-R20))
* Install `mailersend` package as a dependency and update lock file ([link](https://github.com/PipedreamHQ/pipedream/pull/6086/files?diff=unified&w=0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R70), [link](https://github.com/PipedreamHQ/pipedream/pull/6086/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR27), [link](https://github.com/PipedreamHQ/pipedream/pull/6086/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR43), [link](https://github.com/PipedreamHQ/pipedream/pull/6086/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR19929-R19939))

### CODE REVIEW

File: components/mailersend/sources/common/base.mjs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      ],
+    },
+  },
+  methods: {
+    generateMeta(event) {
+      const { data } = event;
+      return {
+        id: data.id,
+        summary: `${data.type}: ${data.email.subject}`.replace(/(^\w)/g, (m) => m.toUpperCase()),
+        ts: +Date.parse(data.created_at),
```
Comment (feyzullah) [inline components/mailersend/sources/common/base.mjs:20]: I could not get what `+` sign does here. Is it for purpose? If not can we remove them?

File: components/mailersend/mailersend.app.mjs [hunk 2]
```diff
@@ -1,11 +1,186 @@
...
   type: "app",
   app: "mailersend",
-  propDefinitions: {},
+  propDefinitions: {
+    domainId: {
+      type: "string",
+      label: "Domain ID",
+      description: "The id of a verified domain.",
+      async options({ prevContext }) {
+        const page = (prevContext?.nextPage || 0);
```
Comment (feyzullah) [inline components/mailersend/mailersend.app.mjs:18]: I could not get what brackets do here. Is it on purpose? If not can we remove them?

**Review by feyzullah — COMMENTED** [review]
Hi @alysonturing, I could not be sure about two things and left comments about them. Can you please check? Because they are not affecting anything, this one is good to go to QA.

File: components/mailersend/sources/common/base.mjs (hunk 1 above)
Comment (alysonturing) [reply components/mailersend/sources/common/base.mjs:20]: This plus serves the purpose of converting the date to timestamp, I haven't tested without it

File: components/mailersend/mailersend.app.mjs (hunk 2 above)
Comment (alysonturing) [reply components/mailersend/mailersend.app.mjs:18]: Sure it can be removed

**Review by feyzullah — APPROVED** [review]

**Comment by vunguyenhung** [comment]
Hi everyone, all test cases are passed! Ready for release! 

Test report
[Mailersend_6086_2200.pdf](https://github.com/PipedreamHQ/pipedream/files/11323888/Mailersend_6086_2200.pdf)

**Pull request merged**